apps/demos/Demos/Chat/AIAndChatbotIntegration/React/App.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from 'react';
+import Chat, { ChatTypes } from 'devextreme-react/chat';
+import { AzureOpenAI } from 'openai';
+import { MessageEnteredEvent } from 'devextreme/ui/chat';
+import CustomStore from 'devextreme/data/custom_store';
+import DataSource from 'devextreme/data/data_source';
+import { loadMessages } from 'devextreme/localization';
+import { 
+  user,
+  assistant,
+  AzureOpenAIConfig,
+  REGENERATION_TEXT,
+  CHAT_DISABLED_CLASS,
+  ALERT_TIMEOUT
+} from './data.ts';
+import Message from './Message.tsx';
+
+const store = [];
+const messages = [];
+
+loadMessages({
+  en: {
+    'dxChat-emptyListMessage': 'Chat is Empty',
+    'dxChat-emptyListPrompt': 'AI Assistant is ready to answer your questions.',
+    'dxChat-textareaPlaceholder': 'Ask AI Assistant...',
+  },
+});
+
+const chatService = new AzureOpenAI(AzureOpenAIConfig);
+
+async function getAIResponse(messages) {
+  const params = {
+    messages,
+    max_tokens: 1000,
+    temperature: 0.7,
+  };
+
+  const response = await chatService.chat.completions.create(params);
+  const data = { choices: response.choices };
+
+  return data.choices[0].message?.content;
+}
+
+function updateLastMessage(text = REGENERATION_TEXT) {
+  const items = dataSource.items();
+  const lastMessage = items.at(-1);
+
+  dataSource.store().push([{
+    type: 'update',
+    key: lastMessage.id,
+    data: { text },
+  }]);
+}
+
+function renderAssistantMessage(text) {
+  const message = {
+    id: Date.now(),
+    timestamp: new Date(),
+    author: assistant,
+    text,
+  };
+
+  dataSource.store().push([{ type: 'insert', data: message }]);
+}
+
+const customStore = new CustomStore({
+  key: 'id',
+  load: () => {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve([...store]);
+      }, 0);
+    });
+  },
+  insert: (message) => {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        store.push(message);
+        resolve(message);
+      });
+    });
+  },
+});
+
+const dataSource = new DataSource({
+  store: customStore,
+  paginate: false,
+})
```

Comment:
Cannot we move everything related to work with dataSource/openAI to a separate hook/hooks to make this file more clear?